src/ops.jl

```diff
@@ -356,3 +356,43 @@ Base.floor{T}(::Type{T}, A::FixedArray) = map(x->floor(T, x), A)
 Base.ceil{T}( ::Type{T}, A::FixedArray) = map(x->ceil( T, x), A)
 Base.trunc{T}(::Type{T}, A::FixedArray) = map(x->trunc(T, x), A)
 Base.round{T}(::Type{T}, A::FixedArray) = map(x->round(T, x), A)
+
+import Base.Cartesian.@nif
+shift(t::Tuple) = Base.tail(t)
+unshift(t::Tuple, val) = tuple(val, t...)
+push(t::Tuple, val) = tuple(t..., val)
+@generated unpush{N,T}(t::NTuple{N,T}) = begin # is there a better name?
```

Comment:
> is there a better name

Good question.  `pop`?  It's inconsistent with `Base.pop!` returning the popped element rather than the container but that's no worse than `shift` which returns the container vs `Base.shift!` which returns the first element.